File: src/main/java/com/fx/billsoft/domain/billnote/service/AccountTypeService.java
package com.fx.billsoft.domain.billnote.service;

import com.fx.billsoft.domain.billnote.entity.AccountType;
import com.fx.billsoft.infrastructure.billnote.repository.AccountTypeRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import java.util.List;
import java.util.Optional;

@Service
public class AccountTypeService {
    private final AccountTypeRepository accountTypeRepository;

    @Autowired
    public AccountTypeService(AccountTypeRepository accountTypeRepository) {
        this.accountTypeRepository = accountTypeRepository;
    }

    public AccountType createAccountType(AccountType accountType) {
        return accountTypeRepository.save(accountType);
    }

    public Optional<AccountType> getAccountTypeById(Long id) {
        return accountTypeRepository.findById(id);
    }

    public AccountType updateAccountType(AccountType accountType) {
        return accountTypeRepository.save(accountType);
    }

    public void deleteAccountType(Long id) {
        accountTypeRepository.deleteById(id);
    }

    public List<AccountType> getAccountTypesByUserId(Long userId) {
        return accountTypeRepository.findByUserId(userId);
    }

    public List<AccountType> getAccountTypesByUserIdAndCategory(Long userId, Integer category) {
        return accountTypeRepository.findByUserIdAndCategory(userId, category);
    }
}

File: frontend/src/services/accountTypeService.js
import http from './http'

class AccountTypeService {
  // 获取所有账目类型
  getAccountTypes() {
    return http.get('/account-types')
  }

  // 根据用户ID获取账目类型
  getAccountTypesByUserId(userId) {
    return http.get(`/account-types/user/${userId}`)
  }

  // 根据用户ID和类别获取账目类型
  getAccountTypesByUserIdAndCategory(userId, category) {
    return http.get(`/account-types/user/${userId}/category/${category}`)
  }

  // 获取账目类型详情
  getAccountType(id) {
    return http.get(`/account-types/${id}`)
  }

  // 创建账目类型
  createAccountType(accountType) {
    return http.post('/account-types', accountType)
  }

  // 更新账目类型
  updateAccountType(id, accountType) {
    return http.put(`/account-types/${id}`, accountType)
  }

  // 删除账目类型
  deleteAccountType(id) {
    return http.delete(`/account-types/${id}`)
  }
}

export default new AccountTypeService()